app_dart/test/model/ci_yaml/target_test.dart

```diff
@@ -290,6 +274,19 @@ void main() {
         expect(dimensions[0].value, 'mac');
       });
 
+      test('target specific dimensions overrides legacy target specific properties', () {
+        final Target target = generateTarget(
+          1,
+          platform: 'Windows',
+          dimensions: <String, String>{'cpu': 'x64'},
+          properties: <String, String>{'cpu': 'x32'},
+        );
+        final List<RequestedDimension> dimensions = target.getDimensions();
+        expect(dimensions.length, 1);
+        expect(dimensions[0].key, 'cpu');
+        expect(dimensions[0].value, 'x64');
+      });
```

Comment:
Can we also add a test for the case of targets being able to override the legacy platform property dimensions

Reply:
for sure! added